vllm/v1/core/sched/scheduler.py

```diff
@@ -434,6 +450,14 @@ def schedule(self) -> SchedulerOutput:
             grammar_bitmask=grammar_bitmask,
         )
 
+        # NOTE(Kuntai): this function is designed for multiple purposes:
+        # 1. Plan the KV cache store
+        # 2. Wrap up all the KV cache load / save ops into an opaque object
+        # 3. Clear the internal states of the connector
+        if self.connector is not None:
+            scheduler_output = self.connector.attach_connector_meta(
+                scheduler_output)
```

Comment:
Can we be more explicit about the change, like below?
```suggestion
            scheduler_output.kv_connector_metadata = self.connector.attach_connector_meta(
                scheduler_output)
```

Reply:
Also, it would be nice if we can clearly specify the requirements to the scheduler output. I think the API will keep evolving (not super frequently, but from time to time) going forward.

Reply:
I think this will vary backend by backend, so its bit difficult to know.

Updated the function to be pure